src/components/Avatar.vue

```diff
@@ -22,13 +22,12 @@ export default {
             return ((this.message && this.message.nick) || (this.user && this.user.nick) || '')[0];
         },
         hasAvatar() {
-            return !!(this.avatar && this.avatar.small);
+            return !!(this.avatar && (this.avatar.small || this.avatar.large));
         },
         avatarStyle() {
             let style = {};
-
             if (this.hasAvatar) {
-                let url = this.avatar.small;
+                let url = (this.size === 'small' && this.avatar.small) ? this.avatar.small : this.avatar.large;
```

Comment:
This should probably get split up to make it easier to read at this point
~~~javascript
let url = (this.size === 'small' && this.avatar.small) ?
    this.avatar.small :
    this.avatar.large;
~~~